addons/mail/static/src/discuss/core/common/thread_model_patch.js

```diff
@@ -255,19 +268,24 @@ const threadPatch = {
             return undefined;
         }
         const correspondents = this.correspondents;
-        if (correspondents.length === 1) {
+        if (correspondents?.length === 1) {
             // 2 members chat.
             return correspondents[0];
         }
-        if (correspondents.length === 0 && this.channel_member_ids.length === 1) {
+        if (correspondents?.length === 0 && this.channel?.channel_member_ids.length === 1) {
             // Self-chat.
-            return this.channel_member_ids[0];
+            return this.channel?.channel_member_ids[0];
```

Comment:
I guess this is the main issue about correspondent, API is changed here too. This should fallback to return empty array and not undefined.